### PR TITLE
Feature/frontend build files cleanup

### PR DESCRIPTION
**Pull request by achen2401**
address [TN-1952](https://jira.movember.com/browse/TN-1952)
address security vulnerability warnings generated when building frontend files.  Run `npm audit` will give details as to what they are and what packages are affected. 
Changes include:
remove unused packages from package json
upgrade gulp to version 4 as version 4 addressed vulnerabilities as listed from running audit
**note**, this is a breaking change as syntax changes are involved for specifying a task to run.
@ivan-c Ivan, do you want to give this a try and see if anything breaks for you?
One thing I had to do is to do a clean up existing installs `npm prune --production` and then remove the node_modules folder completely - before proceeding to do a rebuild.  This is to prevent incompatible packages (i.e. for gulp) to coexist. 
**do not merge yet**

### CODE REVIEW

**Comment by ivan-c** [comment]
@achen2401 When you get a chance, could you fix the merge conflicts? 
Would love to try this out with i18n now :)

Thanks!

**Comment by achen2401** [comment]
@ivan-c Ivan, conflict should be resolved.  I did try this out again in my instance just to make sure the build processes (e.g. frontend files and translations) run successful.  But let me know if you see or run into any issue.

File: portal/development_gulpfile.js [hunk 1]
```diff
@@ -102,101 +76,137 @@ function replaceStd(fileName) {
...
         });
-    return true;
-});
+    callback();
+};
+exports.psaTrackerLess = series(psaTrackerLess);
+
 /*
- * running watch task will update css automatically in vivo
+ * running each watch task will update css automatically in vivo
```
[the remark this answers is not in the document]
Comment (achen2401) [reply portal/development_gulpfile.js]: @ivan-c Ivan, just noting here that I think marking these packages optional won't help as they are part of gulp package's dependencies. (I did try).  But I think the behavior here is **correct** as npm correctly skips these optional dependencies due to incompatible operating system.  see note here about optional dependency warning from gulp website:  https://gulpjs.com/docs/en/getting-started/watching-files  Also noting here that there is discussion about an open ticket to fix these warnings:  https://github.com/ml5js/ml5-library/issues/252  I will make note of these in the JIRA issue as well

[the remark this answers is not in the document]
Comment (ivan-c) [reply portal/development_gulpfile.js]: It's correct by their definition, but not mine (I don't think warnings should be displayed if something is working 100% correctly)

Thank you humoring me and trying to disable it! :)

**Pull request merged**